app/src/ui/delete-branch/delete-branch-dialog.tsx

```diff
@@ -94,12 +95,17 @@ export class DeleteBranch extends React.Component<
     this.setState({ includeRemoteBranch: value })
   }
 
-  private deleteBranch = () => {
-    this.props.dispatcher.deleteBranch(
-      this.props.repository,
-      this.props.branch,
-      this.state.includeRemoteBranch
-    )
+  private deleteBranch = async () => {
+    await Promise.all([
+      this.props.dispatcher.deleteBranch(
+        this.props.repository,
+        this.props.branch,
+        this.state.includeRemoteBranch
+      ),
+      this.props.dispatcher.executeCompare(this.props.repository, {
```

Comment:
This seems really unrelated to the delete branch workflow - can we add a comment here indicating why this is important?